### PR TITLE
[NETBEANS-5245] Show and change the PHP Version of project properties on the status bar

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-5245

- Show a PHP Version of project properties on the status bar
- We click a specific version number of the version list, we can change the PHP Version to it without opening a project properties dialog

netbeans-5245-php-version-on-status-bar

### CODE REVIEW

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (junichi11) [inline php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:244]: @tmysik I'm not sure whether this is OK/correct for you...

**Comment by junichi11** [comment]
I've implemented this based on https://github.com/apache/netbeans/blob/master/ide/editor/src/org/netbeans/modules/editor/impl/crlf/CRLFStatus.java
as the first prototype: https://github.com/junichi11/netbeans/commit/613f5008c261f4dc01ffee5d6a3aa4a174c901ac

I've added the label of PHP Version as a static field in it. Instead of that, I've created `PhpVersionLabel` class in this PR.

**Comment by tmysik** [comment]
> I've implemented this based on https://github.com/apache/netbeans/blob/master/ide/editor/src/org/netbeans/modules/editor/impl/crlf/CRLFStatus.java

Good. This was done by @jlahoda if I am not wrong and this guy knows what he is doing :)

File: php/php.project/nbproject/project.xml
Comment (tmysik) [inline php/php.project/nbproject/project.xml:156]: As discussed in another PR, the specification version of php.project module should be increased too.

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (tmysik) [inline php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:157]: This is done in the UI thread... hmm, hopefully it will be fast, ideally always :)

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (tmysik) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:244]: It would be perhaps better to save it in a background thread...

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (tmysik) [inline php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:241]: Don't we have any utility code for this already? :thinking:

**Review by tmysik — APPROVED** [review]

**Comment by tmysik** [comment]
@junichi11 

In general, I am OK with this change, thanks for it! The only thing - we should really save the project properties in a background thread to avoid blocking of the UI.

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (junichi11) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:244]: Exactly.

File: php/php.project/nbproject/project.xml
Comment (junichi11) [reply php/php.project/nbproject/project.xml:156]: Yes, will fix. Thanks!

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (tmysik) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:157]: (Otherwise some kind of "Loading..." would need to be added.)

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (junichi11) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:241]: Yes, we have `PhpProjectUtils.getPhpProject(FileObject)`. Will fix it. Thanks.

Comment (tmysik) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:241]: Great, thank you.

File: php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java
Comment (junichi11) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:157]: e.g. `setText(Bundle.loading_message())`?

Comment (tmysik) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:157]: I meant - if this is slow and blocking the UI, we would need to:
- show Loading text here
- start a bg thread which would do the work
- once finished, the Loading text would be replaced with the PHP version

Is this what you meant? Or...? Thanks!

Comment (junichi11) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:157]: I see. Thanks!

Comment (tmysik) [reply php/php.project/src/org/netbeans/modules/php/project/ui/PhpVersionStatusLineElementProvider.java:161]: Perfect, thank you!

**Comment by junichi11** [comment]
@tmysik Fixed: https://github.com/apache/netbeans/compare/83533a66fe05bed7e8cad7d280ea554cbe016e1a..b35e55a0d72cd6ad847186c9433310f3fcb79ad4 
Please let me know if I'm wrong. 
Maybe, there is a conflict(spec version) between the other PR and this. I'll rebase this after the other PR is merged. Thanks!

**Comment by tmysik** [comment]
@junichi11 

There are some CI failures - could you please have a look? Feel free to merge this PR if they are unrelated. Thank you!

**Comment by tmysik** [comment]
@junichi11 

I see. Thanks!

**Comment by junichi11** [comment]
@tmysik Thank you for your review!

**Pull request merged**
